src/porepy/numerics/ad/forward_mode.py

```diff
@@ -100,6 +100,56 @@ def __repr__(self) -> str:
         s += " elements"
         return s
 
+    def __getitem__(self, key: slice | np._ArrayLikeInt) -> AdArray:
```

Comment:
AdArrays can now be sliced row-wise.
The getter is clear I think.
The setter allows to set the rows in val and jac using an AdArray of respective size.
It also allows to set val only, if the setter receives a numpy array only, and not an AdArray.